---
layout: page
title: Frequently Asked Questions
---

Thank you for your interest in hosting or teaching our Pangenomics Workshop. Below you will find answers to some frequently asked questions (FAQs) about this curriculum. If the answer to your question doesn’t appear here, please contact us using the Contact link in the righthand lower corner of this page.

* [For Hosts](#hosts)
* [For Instructors](#instructors)

## <a id="hosts"></a> Hosts

### What does this workshop cover? 

The Pangenomics Workshop covers essential skills for pangenomics research, including: 
* Best practices for organizing bioinformatics projects and data
* An introduction to Bash and Python programming languages
* An overview of the history, key concepts, and research application of Pangenomics, including practical examples
* Basic concepts and application of Topological Data Analysis (TDA) in the Pangenomics field
* Using command line tools to process, analyze, and visualize pangenomics data
* Connecting to and using cloud computing resources 

### Who is this workshop for and what experience is needed? 

This workshop is designed for individuals interested in Pangenomics research. Attendees do not need prior experience with the tools covered in the workshop. However, some familiarity with biological concepts, suchs as DNA sequences and gene diversity, is expected. 

### What will learners be able to do by the end of the workshop?

By the end of the workshop, learners will be able to: 

- Download genomic sequences from the NCBI database
- Structure, organize, and document their genomic data and metadata
- Understand what a bioinformatic workflow is
- Use the terminal to navigate, create, copy, move, and remove files and directories
- Create simple and useful scripts and wildcards to automate repetitive tasks
- Use command-line tools to perform genome annotation, homology searches, clustering of gene families and pangenome computing, analysis and visualization
- Apply TDA concepts in the context of Pangenomics to conduct Horizontal Gene Transfer analaysis

### What are the software, hardware, and connectivity requirements for this workshop?

Learners must bring a laptop (not a tablet) with any spreadsheet program installed (e.g., LibreOffice, Microsoft Excel). For those using Windows machine, [Git for Windows](https://gitforwindows.org/) must also be downloaded and installed. There are no additional hardware or software requirements. Learners will need a stable and strong internet connection to work on the remote computing system used in this workshop. For more information, please visit the workshop's [Setup](https://carpentries-incubator.github.io/pangenomics-workshop/setup.html) page.

### Can I use my institution's compute cluster or a different cloud computing resource for this workshop?
To ensure a consistent experience for learners and instructors, all centrally-organized workshops by The Carpentries use our stable, community-tested curriculum and technical setup. If you wish to use a different platform for a self-organized workshop, our materials are publicly available and licensed under [CC-BY](https://creativecommons.org/licenses/by/4.0/). For more information about centrally-organized versus self-organized workshops and the use of The Carpentries brand, see the [Teaching and Hosting](https://docs.carpentries.org/topic_folders/hosts_instructors/index.html) section of The Carpentries Handbook.

### Where can I find the complete content of this workshop?
For a full description of this workshop, including the content covered and the dataset used, visit the [Pangenomics Workshop Overview](https://carpentries-incubator.github.io/pangenomics-workshop/)) page. 

## <a id="instructors"></a> Instructors

### What background and technological skills do I need to teach this workshop?

Instructors need experience using Bash shell (the default shell on Mac OS and most Linux systems), including writing small Bash scripts and running programs from the command line. Specific experience with the command-line programs used in the lessons is not required. Prior experience logging into remote computers is useful. Instructors should have experience working with genoic sequences in fastq format. 

### How can I prepare to teach this material? 

Each lesson includes Instructor Notes that provide information about the lesson design, common problems, and technical tips. Access the [Instructor Notes](https://carpentries-incubator.github.io/pangenomics-workshop/guide/index.html). These notes are written collaboratively by our instructors, so please contribute your insights after your workshop!


### How will my learners get a connection and log-in information? 

The Carpentries Workshop and Logistics Manager will provide Instructors with connection information for AMI instances the day before the workshop. Enough instances will be provided for each learner and, if requested, for each workshop helper to have their own individual instance. Instructors can make connection information available to learners through the workshop Etherpad. 

### What are common problems that arise during this workshop?

The best place to get information about common problems that arise during workshops is in the [Instructor Notes](https://carpentries-incubator.github.io/metagenomics-workshop/guide/index.html). You can access Instructor Notes through the Extras menu in the top navigation bar that appears across the head of each lesson. Instructors are strongly encouraged to contribute back to the Instructor Notes based on their workshop experience. To contribute to the Instructor Notes, click the “Improve this page” menu option in the upper right corner of the Instructor Notes page. 

### Does the AWS image location matter? Should I set up an AMI in a different region if my workshop will be held outside of the Eastern US?

We have run this workshop in locations across the United States and Europe with no noticeable difference in instance speed if you experience any issues, please [let us know](team@carpentries.org).

# Reference
These notes are adapted from the Data Carpentry Genomics Workshop [corresponding page](https://datacarpentry.org/genomics-workshop/faq/index.html)
