---
layout: page
title: Frequently Asked Questions
---

Thank you for your interest in hosting or teaching our Pangenomics Workshop. Below you will find answers to some frequently asked questions (FAQs) about this curriculum. If the answer to your question doesnâ€™t appear here, please contact us using the Contact link in the righthand lower corner of this page.

* [For Hosts](#hosts)
* [For Instructors](#instructors)

## <a id="hosts"></a> Hosts

### What does this workshop cover? 

The Pangenomics Workshop covers essential skills for pangenomics research, including: 
* Best practices for organizing bioinformatics projects and data
* An introduction to Bash and Python programming languages
* An overview of the history, key concepts, and research application of Pangenomics, including practical examples
* Basic concepts and application of Topological Data Analysis (TDA) in the Pangenomics field
* Using command line tools to process, analyze, and visualize pangenomics data
* Connecting to and using cloud computing resources 

### Who is this workshop for and what experience is needed? 

This workshop is designed for individuals interested in Pangenomics research. Attendees do not need prior experience with the tools covered in the workshop. However, some familiarity with biological concepts, such as DNA sequences and gene diversity, is expected. 

### What will learners be able to do by the end of the workshop?

By the end of the workshop, learners will be able to: 

- Download genomic sequences from the NCBI database
- Structure, organize, and document their genomic data and metadata
- Understand what a bioinformatic workflow is
- Use the terminal to navigate, create, copy, move, and remove files and directories
- Create simple and useful scripts and wildcards to automate repetitive tasks
- Use command-line tools to perform genome annotation, homology searches, clustering of gene families and pangenome computing, analysis and visualization
- Apply TDA concepts in the context of Pangenomics to conduct Horizontal Gene Transfer analaysis

### What are the software, hardware, and connectivity requirements for this workshop?

Learners must bring a laptop (not a tablet) with any spreadsheet program installed (e.g., LibreOffice, Microsoft Excel). For those using Windows machine, [Git for Windows](https://gitforwindows.org/) must also be downloaded and installed. There are no additional hardware or software requirements. Learners will need a stable and strong internet connection to work on the remote computing system used in this workshop. For more information, please visit the workshop's [Setup](https://carpentries-incubator.github.io/pangenomics-workshop/setup.html) page.

### Can I use my institution's compute cluster or a different cloud computing resource for this workshop?
To ensure a consistent experience for learners and instructors, all centrally-organized workshops by The Carpentries use our stable, community-tested curriculum and technical setup. If you wish to use a different platform for a self-organized workshop, our materials are publicly available and licensed under [CC-BY](https://creativecommons.org/licenses/by/4.0/). For more information about centrally-organized versus self-organized workshops and the use of The Carpentries brand, see the [Teaching and Hosting](https://docs.carpentries.org/topic_folders/hosts_instructors/index.html) section of The Carpentries Handbook.

### Where can I find the complete content of this workshop?
For a full description of this workshop, including the content covered and the dataset used, visit the [Pangenomics Workshop Overview](https://carpentries-incubator.github.io/pangenomics-workshop/) page. 

## <a id="instructors"></a> Instructors

### What background and technological skills do I need to teach this workshop?

Instructors need experience using Bash shell (the default shell on Mac OS and most Linux systems), including writing small Bash scripts and running programs from the command line. Specific experience with the command-line programs used in the lessons is not required. Prior experience logging into remote computers is useful. Instructors should have experience working with genomic sequences in fastq format. 

### How can I prepare to teach this material? 

Each lesson includes [Instructor Notes](https://carpentries-incubator.github.io/pangenomics-workshop/guide/index.html) that provide information about the lesson design, common problems, and technical tips. These notes are written collaboratively by our instructors, so please contribute your insights after your workshop!


### How can I run this workshop for review and practice before teaching it? 

To review and practice this workshop before teaching, refere to the [Setup](https://carpentries-incubator.github.io/pangenomics-workshop/setup.html) section for detailed instructions. You have two main options: 
- **Data Carpentry Workshop.** You can request access to a local server from CCM UNAM by contacting nselem@matmor.unam.mx to obtain user credentials. The server has all the necessary software pre-installed, allowing you to follow the lesson seamlessly.
- **Self-run.** You can set up the workshop environment yourself by manually installing all dependencies or using a [Docker image](https://hub.docker.com/repository/docker/aapashkov/panworkshop) that includes everything you need.

Additionally, the setup page provides a [Zenodo link](https://zenodo.org/records/11374283) where you can download all the genomic data used in the lesson.

### What are common problems that arise during this workshop?

The best source of information about common problems during workshops is the [Instructor Notes](https://carpentries-incubator.github.io/pangenomics-workshop/guide/index.html). Access these notes through the "Extras" menu in the top navigation bar of each lesson. Instructors are encouraged to contribute to the Instructor Notes based on their experiences. To contribute, click the "Improve this page" option in the upper right corner of the Instructor Notes page. 


# Reference
These notes are adapted from the Data Carpentry Genomics Workshop [corresponding page](https://datacarpentry.org/genomics-workshop/faq/index.html)
